src/dc_tools.rs

```diff
@@ -341,19 +339,24 @@ pub unsafe fn dc_utf8_strlen(s: *const libc::c_char) -> size_t {
 
     j
 }
+pub fn dc_truncate_n_str(buf: &str, approx_chars: usize, do_unwrap: bool) -> Cow<str> {
```

Comment:
`usize` is perhaps a bit weird as there are no pointers involved here?  I'd go with `u32`